src/libmain/progress-bar.cc

```diff
@@ -193,15 +196,21 @@ class ProgressBar : public Logger
         }
 
         if (type == actPostBuildHook) {
-            auto name = storePathToName(getS(fields, 0));
+            // String needs to outlive the string_view
+            // https://www.foonathan.net/2017/03/string_view-temporary/
+            std::string s = getS(fields, 0);
+            auto name = storePathToName(s);
             if (hasSuffix(name, ".drv"))
                 name = name.substr(0, name.size() - 4);
             i->s = fmt("post-build " ANSI_BOLD "%s" ANSI_NORMAL, name);
             i->name = DrvName(name).name;
         }
 
         if (type == actQueryPathInfo) {
-            auto name = storePathToName(getS(fields, 0));
+            // String needs to outlive the string_view
+            // https://www.foonathan.net/2017/03/string_view-temporary/
+            std::string s = getS(fields, 0);
+            auto name = storePathToName(s);
```

Comment:
I must admit that I was not very familiar with these new C++11 classes.
I had to read up on what was actually going on here but basically `getS(fields, 0)` is returning a temporary _std::string_ which is alive guaranteed for the whole expression -- so the _string_view_ inside of `getS` is OK.

Unfortunately, `getS` is grabbing a _string_view_ and persisting it in `name` past the lifetime of it resulting in the error.